ui/src/app/views/workspace-share/component.ts

```diff
@@ -176,14 +176,18 @@ export class WorkspaceShareComponent implements OnInit {
       return;
     }
     this.userService.user(this.searchTerm).subscribe((response) => {
-      this.autocompleteLoading = false;
       this.userResponse = response;
-      response.users = response.users.filter(user => {
-        return this.checkUnique(user.email, user.familyName, user.givenName);
-      });
-      this.autocompleteUsers = response.users.splice(0, 4);
-      if (this.autocompleteUsers.length === 0) {
-        this.autocompleteNoResults = true;
+      if (this.searchTerm === this.userResponse.query) {
```

Comment:
This could simply be capture via closure above.

```
const searchTerm = this.searchTerm;
this.userService.user(searchTerm).sub(()=>{
  if (this.searchTerm !== searchTerm) return;
...
});
```

Alternatively, a more general solution:

```
const obs = this.userService.user(... => {
  if (this.pendingAutocomplete !== obs) return;
...
  this.pendingAutocomplete = null;
});
this.pendingAutocomplete = obs;
```

(also could replace "autocompleteLoading" with a helper to check null'ness on the observable).

Better would be to cancel the old obesrvable as well, but since we're moving to promises with the React changes I wouldn't touch that part.

Reply:
Sorry, I don't follow your suggestions here. The point of comparing the last term the user searched with what is in the response allows the UI to perform an update IFF the response matches the search.

Reply:
Yes, that is what my proposed code is doing. A client is able to store which request corresponds to which response either as an instance property or via a closure. Which part is not making sense?

Reply:
I don't see how this would work, but I'll spend some time with the approaches to see if I can grok them.